24_clustering.Rmd

```diff
@@ -17,9 +18,122 @@ features/samples that share pattern.  For example, with clustering, we
+amongst others. The first thing to do when using this package is to load 
+it. It's also possible to use `transformCounts` to transform the data 
+for the clustering.
+```{r load-bluster}
+# Load dependencies
+if (!require(bluster)) {
+  install.packages("bluster")
+  library(bluster)
+}
+
+# Optional: Apply transformation
+tse <- transformCounts(tse, method = "relabundance")
+```
+
+The clustering can be done on the features or the samples.
+
+### Samples clustering
+
+To cluster the samples, we simply need to transpose the assay so that 
+the samples are in rows.
+```{r sample-clust-prep}
+assay <- t(assay(tse, "relabundance"))
+```
+
+Then, to do the clustering, use the `clusterRows` function. Depending 
+on its parameters, it will perform a different algorithm. Here, we'll 
+perform a hierarchical clustering.
+```{r bluster-sample-hclust1}
+# Simple use of the hierarchical clustering which sets the cut height to
+# half the dendrogram height.
+hclust.out <- clusterRows(assay, HclustParam())
+
+# Add data to the TSE
+colData(tse)$clusters <- hclust.out
+
+# Checking the result
+hclust.out
+```
+
+Once the clustering on the samples is done, we can also plot the clusters.
+```{r bluster-sample-plot}
+library(scater)
+# Add the MDS dimensions for plotting
+tse <- runMDS(tse,
+  assay.type = "relabundance", FUN = vegan::vegdist,
+  method = "bray"
+)
+
+# Plot the clusters
+plotReducedDim(tse, "MDS", colour_by = "clusters")
+```
+
+To change the clustering parameters, simply change the parameters in 
+`HclustParam` function. To see the different parameters, check the 
+[HclustParam documentation](https://rdrr.io/github/LTLA/bluster/man/HclustParam-class.html).
+```{r bluster-sample-hclust2}
+# More complex use of the algorithm with another method and cut height.
+# We also add the full parameter to get the full hierarchical clustering
+# information to plot the dendrogram
+hclust.out <- clusterRows(assay, HclustParam(method = "complete"), full = TRUE)
+hclust.out$clusters
+
+# Add data to the TSE
+colData(tse)$clusters <- hclust.out$clusters
+
+# Get the dendrogram object
+dendro <- as.dendrogram(hclust.out$objects$hclust)
+
+# Plot the dendrogram
+plot(dendro)
+```
+
+Alternatively, it's also possible to use the `clusterCells` function from 
+the *scran* package.This function uses directly the tree as a parameter 
+instead of the assay itself. Here, we will use the PAM algorithm.
+```{r scran-sample-clust}
+if (!require(scran)) {
+  BiocManager::install("scran")
+  library(scran)
+}
+pam.out <- clusterCells(tse,
+  assay.type = "relabundance",
+  BLUSPARAM = PamParam(centers = 5)
+)
+```
+
+### Taxa clustering
+
+It's also possible to do clustering for the taxa. We'll do once again the 
+hierarchical clustering.
+```{r taxa-clust}
+# Transform data with CLR
+tse <- transformCounts(tse,
+  assay.type = "relabundance",
+  method = "clr",
+  pseudocount = min(assay(tse, "relabundance")[assay(tse, "relabundance") > 0])
```

Comment:
usually pseudocount is by default minimum value divided by 2. To make a distinction.